### ISSUE TITLE
Unserialize doesn't throw error, returns false instead

### ISSUE THREAD

**Issue by mattKendon** [jgrossi/corcel#102]
This is a wierd one as it appears to have only just started occurring, or I've been otherwise missing this until now. 

The `PostMeta` class has this useful method

```
public function getValueAttribute()
{
    try {
        return unserialize($this->meta_value);
     } catch (Exception $ex) {
         return $this->meta_value;
     }
}
```

However I have just started getting bad data being returned from this as `unserialize` no longer throws an error but instead just returns `false` when it cannot unserialize a value. `php.net` says that `unserialize` should return `false` when unable to unserialize a value. 

When I tried testing this out using the test suite with the corcel library, I couldn't get the same results. As I have seen both, I suggest the following amendment to the code to handle cases like mine where `unserialize` does not play ball. 

```
public function getValueAttribute()
    {
        try {
            $value = unserialize($this->meta_value);
            // if we get false, but the original value is not false then something has gone wrong. 
            // return the meta_value as is instead of unserializing
            // added this to handle cases where unserialize doesn't throw an error that is catchable
            return $value === false && $this->meta_value !== false ? $this->meta_value : $value;
        } catch (Exception $ex) {
            return $this->meta_value;
        }
    }
```

This now checks for a value of `false` being returned from `unserialize` and if so, just checks that against the meta_value. This handles those cases where an Error is not being thrown as something catchable. 

However, I realise that this may not be the best solution and would be happy to hear what others had to say on the matter.

Is anyone else experiencing this?

**Comment by josec89** [comment]
Awesome! 

This might have happened because the `unserialize` method changed from one php version to another or something like that.

I like your approach, lets wait to see what *big boss* says :)

**Comment by jgrossi** [comment]
@josec89 Big boss hahahahahahaha

Hi @mattKendon thanks for the report! I think it's a very nice approach. I'm gonna start to create some Exceptions classes and maybe this can be one in this case.

Just send the PR to dev branch ;-) Thanks for the contribution!

**Comment by mattKendon** [comment]
Created a PR for this and to also fix the same issue on the Options class #103

**Comment by jgrossi** [comment]
Done!
